### PR TITLE
feat: Add arrayOf rule

### PR DESCRIPTION
**Pull request by relekang**
The rule takes config object for a new validator as options and all items
in the array will be validated based on that config.

### CODE REVIEW

File: src/rules/array.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { keys, map } from 'lodash';
+import Validator from '../validator';
+
+export function arrayOf(field, value, options) {
+  console.log(options);
```
Comment (marteloge) [inline src/rules/array.js]: Remove console.log?

Comment (relekang) [inline src/rules/array.js]: Fixed

File: src/rules/array.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { keys, map } from 'lodash';
+import Validator from '../validator';
+
+export function arrayOf(field, value, options) {
+  const validator = new Validator(options);
+  return map(value, item => {
+    const result = validator.validate(keys(options), item);
```
Comment (marteloge) [inline src/rules/array.js]: return validator.validate(keys(options), item); ?

Comment (relekang) [inline src/rules/array.js]: 👍  Fixed

**Comment by marteloge** [comment]
Ship it! :)

**Pull request merged**